lib/ansible/modules/include_vars.py

```diff
@@ -70,6 +70,13 @@
     type: bool
     default: no
     version_added: "2.7"
+  merge_hashes:
+    description:
+      - Merges existing hash variables instead of overwriting them.
+      - When false, the behavior falls back to the global 'hash_behaviour' configuration.
```

Comment:
actually, after looking at code 'currently', include_vars always ovewrites as it does not use combine_vars at any point ..  this might be a 'bug' in VarsManager on `set_host_facts` and `set_nonpersistent_facts` methods .. unsure if we need to fix considering upcoming deprecation.

Reply:
> include_vars always ovewrites as it does not use combine_vars at any point

`hash_behaviour = merge` works as expected with `include_vars` and we've been relying on that behavior.

```
───────┬────────────────────────────────────────────────────────────────────────
       │ File: ansible.cfg
───────┼────────────────────────────────────────────────────────────────────────
   1   │ [defaults]
   2   │ hash_behaviour = merge
   3   │ nocows = 1
───────┴────────────────────────────────────────────────────────────────────────
───────┬────────────────────────────────────────────────────────────────────────
       │ File: a.yml
───────┼────────────────────────────────────────────────────────────────────────
   1   │ foo: { bar: 1 }
───────┴────────────────────────────────────────────────────────────────────────
───────┬────────────────────────────────────────────────────────────────────────
       │ File: b.yml
───────┼────────────────────────────────────────────────────────────────────────
   1   │ foo: { baz: 2 }
───────┴────────────────────────────────────────────────────────────────────────
───────┬────────────────────────────────────────────────────────────────────────
       │ File: hosts
───────┼────────────────────────────────────────────────────────────────────────
   1   │ localhost
───────┴────────────────────────────────────────────────────────────────────────
───────┬────────────────────────────────────────────────────────────────────────
       │ File: main.yml
───────┼────────────────────────────────────────────────────────────────────────
   1   │ - hosts: all
   2   │   gather_facts: no
   3   │   tasks:
   4   │     - include_vars: a.yml
   5   │     - include_vars: b.yml
   6   │     - debug:
   7   │         var: foo
───────┴────────────────────────────────────────────────────────────────────────
```

```
PLAY [all] ***********************************************************************************************************

TASK [include_vars] **************************************************************************************************
ok: [localhost]

TASK [include_vars] **************************************************************************************************
ok: [localhost]

TASK [debug] *********************************************************************************************************
ok: [localhost] => {
    "foo": {
        "bar": 1,
        "baz": 2
    }
}
```

If we comment out `hash_behaviour`,

```
TASK [debug] *********************************************************************************************************
ok: [localhost] => {
    "foo": {
        "baz": 2
    }
}
```